### PR TITLE
kdePackages.kdeconnect-kde: hardcode sshfs path

### LINKED ISSUE #311236: KDE Connect cannot browse device in Plasma 6
### Describe the bug
I cannot browse my android device with KDE Connect in Plasma 6. If I try to open my phone in Dolphin, I receive the following error message.
```
Error when accessing filesystem. sshfs finished with exit code 1 
```

When I compare the nix package definition for [`libsForQt5.kdeconnect-kde`](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/kde/kdeconnect-kde.nix#L77) and [`kdePackages.kdeconnect-kde`](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/kde/gear/kdeconnect-kde/default.nix#L12), I notice that the older one manually add `sshfs` in the `PATH`:

https://github.com/NixOS/nixpkgs/blob/062ca2a9370a27a35c524dc82d540e6e9824b652/pkgs/applications/kde/kdeconnect-kde.nix#L72-L74

I'm wondering if this could be the source of issue.

### Steps To Reproduce
1. Browse device using KDE Connect via Dolphin in Plasma 6

### Expected behavior
I should be able to browse my device in Dolphin

### Additional context
If I remember correctly KDE Connect worked for me in Plasma 5.

### Notify maintainers
@K900 

### Metadata
```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.8.9, NixOS, 24.05 (Uakari), 24.05.20240511.062ca2a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/pfc56yr7y3wflvbgnrpscf2n1m4j3xd7-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by K900**
Fixes #311236

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/kde/gear/kdeconnect-kde/remove-old-host-key-algorithms.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff --git a/plugins/sftp/mounter.cpp b/plugins/sftp/mounter.cpp
+index 29e94f3b..802ba679 100644
+--- a/plugins/sftp/mounter.cpp
++++ b/plugins/sftp/mounter.cpp
+@@ -122,7 +122,6 @@ void Mounter::onPacketReceived(const NetworkPacket &np)
+                       << QStringLiteral("-o") << QStringLiteral("IdentityFile=") + KdeConnectConfig::instance().privateKeyPath()
+                       << QStringLiteral("-o") << QStringLiteral("StrictHostKeyChecking=no") // Do not ask for confirmation because it is not a known host
+                       << QStringLiteral("-o") << QStringLiteral("UserKnownHostsFile=/dev/null") // Prevent storing as a known host
+-                      << QStringLiteral("-o") << QStringLiteral("HostKeyAlgorithms=+ssh-dss\\,ssh-rsa") // https://bugs.kde.org/show_bug.cgi?id=351725
```
Comment (SuperSandro2000) [inline pkgs/kde/gear/kdeconnect-kde/remove-old-host-key-algorithms.patch:9]: ssh-dss is not supported by our openssh, but ssh-rsa is

File: pkgs/kde/gear/kdeconnect-kde/default.nix [hunk 2]
```diff
@@ -11,6 +14,19 @@
...
+    (substituteAll {
+      src = ./hardcode-sshfs-path.patch;
+      sshfs = lib.getExe sshfs;
+    })
+  ];
+
+  # Hardcoded as a QString, which is UTF-16 so Nix can't pick it up automatically
+  postFixup = ''
+    mkdir -p $out/nix-support
+    echo "${sshfs}" > $out/nix-support/depends
```
Comment (SuperSandro2000) [inline pkgs/kde/gear/kdeconnect-kde/default.nix:30]:
```suggestion
    echo "${sshfs}" >> $out/nix-support/depends
```

Comment (K900) [reply pkgs/kde/gear/kdeconnect-kde/default.nix:31]: That file isn't real and is not created by anything except us.

File: pkgs/kde/gear/kdeconnect-kde/remove-old-host-key-algorithms.patch (hunk 1 above)
Comment (K900) [reply pkgs/kde/gear/kdeconnect-kde/remove-old-host-key-algorithms.patch]: Updated the patch and the comment.

File: pkgs/kde/gear/kdeconnect-kde/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/kde/gear/kdeconnect-kde/default.nix:31]: ah okay, I thought that was somehow also a special nix file.

**Pull request merged**